First Attempt at To-Do List

ibs-alt opened the pull request:
Currently able to add and view the list of  inputted tasks to the To-Do List, and exit the program.

Next iteration will look at updating task status and deleting tasks once complete.

ibs-alt committed:
First Attempt at To-Do List

Currently able to add and view the list of  inputted tasks to the To-Do List, and exit the program.

Next iteration will look at updating task status and deleting tasks once complete.

ibs-alt committed:
Added Delete Task Feature

The user now has the option to delete a task from the task list.

First, it checks if there are any tasks to delete then displays the current tasks with numbers for reference and gets the task number to delete from the user. The deleetion happens using the slice operation

ibs-alt committed:
Implemented Task Update + Task Deletion

AdamGreenhillBJSS commented on Todo1.go:
```diff
@@ -0,0 +1,94 @@
+// Package main implements a simple command-line todo list application
```
Is this file pretty much superceded by 'main.go' at this point? Can you remove it if so?

AdamGreenhillBJSS commented on main.go:
```diff
@@ -0,0 +1,123 @@
+	"strings"
+)
+
+type Todo struct { // Todo represents a single task with a description and status.
```
It's worth adding an ID field to this struct so you have something fixed to refer to, rather than relying on the implicit ordering of the JSON file

AdamGreenhillBJSS commented on main.go:
```diff
@@ -0,0 +1,123 @@
+	Status      string `json:"status"`
+}
+
+const fileName = "data.json"
```
Just to prove you know how to use the `flags` package, could you make this file name configurable as a CLI flag?

AdamGreenhillBJSS commented on main.go:
```diff
@@ -0,0 +1,123 @@
+	data, err := os.ReadFile(fileName)
+	if err == nil {
+		_ = json.Unmarshal(data, &todos)
+	}
```
A couple points on error handling:

1. It's conventional to have the main flow outside of code blocks where possible, with the error check being inside a block. So in this case you should have an `if err != nil` block rather than a `if err==nil` one.
2. In general, please don't ignore errors. No need to to anything fancy but at a minimum you should log the error, and consider whether to recover or exit the program. In this specific case it's probably a time that you need to exit the program.

AdamGreenhillBJSS commented on main.go:
```diff
@@ -0,0 +1,123 @@
+		fmt.Println("3. Update Task Status")
+		fmt.Println("4. Delete Task")
+		fmt.Println("5. Exit")
+		fmt.Print("-> ")
```
Go's multi-line string format would be useful here. You can enclose a string with backticks and then define this whole thing as one block going over multiple lines. I think something like this but don't quote me on it:
```go
fmt.Println(`Choose an option:
1. Add Task
2. List Tasks
3. etc
`)
```

ibs-alt committed:
Delete Todo1.go

Superceded by main.go

ibs-alt committed:
Create README.md

ibs-alt committed:
Improvements

ID field added to struct

Flags package implementation using go run . --filename=data.json. When I run go run . --file=alt.json the program uses that file instead. This shows the flag works and overrides the default as expected so the user can manage multiple task lists without modifying the code.

The error handling has been amended to follow Go appropriate standards which helps bug fixes

Raw string menu has cleaned up the menu formatting
